Add more explicit annotations of top level coexistence

JustinPihony opened the pull request:
A user recently ran into some difficulties when trying to convert a provided classic system and then create a top level typed actor. The idioms changed between classic and typed, so it should be called out and documented explicitly for transitions such as this.

JustinPihony committed:
Add more explicit annotations of top level coexistence

JustinPihony committed:
Re-added space

JustinPihony committed:
Off by one folder

johanandren commented on akka-docs/src/main/paradox/typed/coexisting.md:
```diff
@@ -87,7 +87,15 @@ Java
 @scala[That adds some implicit extension methods that are added to classic and typed `ActorSystem`, `ActorContext` and `ActorRef` in both directions.]
 @java[To convert between typed and classic `ActorSystem`, `ActorContext` and `ActorRef` in both directions there are adapter methods in `akka.actor.typed.javadsl.Adapter`.]
-Note the inline comments in the example above.
+Note the inline comments in the example above. Additionally, note that the above method of using a top level classic actor is the suggested path for this type of co-existence. However, if you prefer to start with a typed top level actor then you can use the implicits and then call `spawn` directly from the typed system:
```
```suggestion
Note the inline comments in the example above. 

This method of using a top level classic actor is the suggested path for this type of co-existence. However, if you prefer to start with a typed top level actor then you can use the @scala[implicit `spawn` -method]@java[`Adapter.spawn`] directly from the typed system:
```

(I didn't test render this with paradox though)

johanandren commented on akka-docs/src/main/paradox/typed/coexisting.md:
```diff
@@ -87,7 +87,15 @@ Java
+Java
+:  @@snip [TypedWatchingClassicTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/coexistence/TypedWatchingClassicTest.java) { #create }
+
+Once migrated to a fully typed system the concept is to provide the root actor in the `ActorSystem` constructor, and then spawn actors from the `ActorContext` of the actor itself. The rationale for this is partly about consistency. In a typed system you can’t create children to an arbitrary actor from anywhere in your app without messaging it, so this will also hold true for the User guardian act ...
```
Perhaps flesh out https://doc.akka.io/docs/akka/current/typed/from-classic.html#actorsystem and link to that instead?

JustinPihony replied:
Sorry for the long delay here. I'm unsure to what you are referring. Are you suggesting to put this last paragraph into the from-classic because it is more related to that, or the whole thing? I can understand the former so will move in that direction, but let me know if you meant otherwise.

JustinPihony replied:
Moved - and removed "Whereas in a classic system it was mostly not noticed that the guardian was an actor." as it seemed superfluous.

JustinPihony committed:
Accept suggestion

Co-authored-by: Johan Andrén <johan@markatta.com>

JustinPihony committed:
Move documentation to a different section

Removed "Whereas in a classic system it was mostly not noticed that the guardian was an actor." as it is unnecessary

JustinPihony committed:
Final change from comments

JustinPihony committed:
Removed double "this"

JustinPihony commented:
This is now ready for re-review please :)

octonato reviewed (approved):
LGTM

Thanks @JustinPihony
